src/components/Button/Button.styles.ts

```diff
@@ -0,0 +1,127 @@
+import { Button as MuiButton, CircularProgress } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import {
+  blueScale,
+  redScale,
+  greenScale,
+  greyScale,
+} from '../../style-library/types/theme.helpers';
+
+import { SIZE_CONFIGS, SHAPE_CONFIGS } from './Button.constants';
+
+import type { StyledButtonProps } from './Button.types';
+
+// Map variant → palette
+const VARIANT_COLORS = {
+  primary: {
+    main: blueScale[600],
+    light: blueScale[50],
+    hover: blueScale[700],
+    active: blueScale[800],
+  },
+  secondary: {
+    main: greyScale[600],
+    light: greyScale[50],
+    hover: greyScale[700],
+    active: greyScale[800],
+  },
+  success: {
+    main: greenScale[600],
+    light: greenScale[50],
+    hover: greenScale[700],
+    active: greenScale[800],
+  },
+  error: { main: redScale[600], light: redScale[50], hover: redScale[700], active: redScale[800] },
+} as const;
+
+export const StyledButton = styled(MuiButton, {
+  shouldForwardProp: (prop) => !prop.toString().startsWith('$'),
+})<StyledButtonProps>(({ theme, $variant, $size, $shape, $loading, $fullWidth }) => {
+  const colors = VARIANT_COLORS[$variant];
+  const size = SIZE_CONFIGS[$size];
+  const shape = SHAPE_CONFIGS[$shape];
+
+  return {
+    height: size.height,
+    minWidth: $fullWidth ? '100%' : size.minWidth,
+    padding: size.padding,
+    fontSize: size.fontSize,
+    fontWeight: 500,
+    textTransform: 'none',
+    borderRadius: shape.borderRadius,
+    border: 'none',
+    position: 'relative',
+    overflow: 'hidden',
+
+    // contained
+    backgroundColor: colors.main,
+    color: '#fff',
+    boxShadow: theme.shadows[2],
+    cursor: $loading ? 'default' : 'pointer',
+
+    '&:hover': {
+      backgroundColor: colors.hover,
+      boxShadow: theme.shadows[4],
+      transform: 'translateY(-1px)',
+    },
+    '&:focus': {
+      outline: 'none',
+      boxShadow: `${theme.shadows[4]}, 0 0 0 3px ${colors.light}`,
+    },
+    '&:active': {
+      backgroundColor: colors.active,
+      transform: 'translateY(0)',
+      boxShadow: theme.shadows[1],
+    },
+
+    '&:disabled': {
+      backgroundColor: greyScale[300],
```

Comment:
Disabled backgroundColor uses greyScale[300] and color uses greyScale[500] (hard-coded color tokens). These should map to theme.palette.action.disabledBackground / theme.palette.text.disabled or use theme palette tokens.

**Sugerencia:**
```typescript
Use theme tokens: backgroundColor: theme.palette.action.disabledBackground, color: theme.palette.text.disabled
```